keps/sig-node/2570-memory-qos/README.md

```diff
@@ -104,6 +107,62 @@ Brief map as follows:
 | pod | sum(requests.memory) | n |
 | node | n/a | pods, kube-reserved, system-reserved | n |
 
+#### Beta
+The formula for memory.high is modified in Beta version of the feature. It will be set based on formula:
+```
+memory.high=floor[(requests.memory + memory throttling factor * (limits.memory or node allocatable memory - requests.memory))/pageSize] * pageSize, where default value of memory throttling factor is set to 0.9
+```
+
+e.g. If a container has `requests.memory=50, limits.memory=100`, and we have a throttling factor of .9, `memory.high` would be 90. If a container had no memory limit specified, we substitute `limits.memory` for `node allocatable memory` and apply the throttling factor of .9 to that value.
+
+Node reserved resources(kube-reserved/system-reserved) are either considered. It is tied to `--enforce-node-allocatable` and `memory.min` will be set properly.
+
+Brief map as follows:
+| type | memory.min | memory.high |
+| -------- | -------- | -------- | 
+| container | requests.memory | floor[(requests.memory + memory throttling factor * (limits.memory or node allocatable memory - requests.memory))/pageSize] * pageSize |
+| pod | sum(requests.memory) | n |
+| node | n/a | pods, kube-reserved, system-reserved | n |
+
+
+**Note:** The formula for memory.high is changed in Beta as the Alpha implementation has following problems:
+1. It fails to throttle when requested memory is closer to memory limits (or node allocatable): memory.high < requests.memory. e.g. If `requests.memory = 85, limits.memory=100`, and we have a throttling factor of 0.8, then as per the Alpha implementation memory.high =  memory throttling factor * limits.memory i.e. memory.high = 80. In this case the level at which throttling is supposed to occur i.e. memory.high is less than requests.memory. Hence there won't be any throttling as the Alpha implementation doesn't allow memory.high to be less than requested memory. 
+
+2. Default throttling factor of 0.8 might be too aggressive for some applications that are latency sensitive and always use memory close to memory limits.
+
+For detailed reasoning around the change, read the documentation [`here`](https://docs.google.com/document/d/1r-e_jWL5EllBgxtaNb9qI2eLOKf9wSuuuglTySWmfmQ/edit?usp=sharing).
+
+
+**Quality of Service for Pods**
+In addition to the change in formula for memory.high, we are also adding the support for memory.high to be set as per `Quality of Service(QOS) for Pod` classes. Following are the different cases:
+
+1. Guaranteed
+The pod gets a Guaranteed class if the request and limit values are the same. Guaranteed class pods are the highest priority pods and throttling is disabled for these pods by not setting memory.high thereby allowing these pods to use the maximum available/allocated memory.
```

Comment:
nit: I would add one more sentence here to explain _why_ we decided to do this. Maybe something like:

Guaranteed pods by their QoS definition require `memory requests= memory limits` and are not overcommitted, so MemoryQoS feature is disabled on those pods, by not setting `memory.high`. This ensures that Guaranteed pods can fully use their memory requests and not hit any throttling.